packages/medusa/src/api/store/products/helpers.ts

```diff
@@ -8,3 +14,96 @@ export const refetchProduct = async (
 ) => {
   return await refetchEntity("product", idOrFilter, scope, fields)
 }
+
+type VariantInventoryType = {
+  variant_id: string
+  variant: { manage_inventory: boolean }
+  required_quantity: number
+  inventory: InventoryItemDTO
+}
+
+export const wrapVariantsWithInventoryQuantity = async (
```

Comment:
The function feels a bit overwhelming, so maybe we can reduce complexity by splitting the code up a bit? Not a blocker for sure, just something I felt while reading the code.

Reply:
Any recommendations? Most of the function is just gathering data and a few lines of code to calculate